modules/va_forms/app/controllers/va_forms/v0/forms_controller.rb

```diff
@@ -6,7 +6,44 @@ class FormsController < ApplicationController
       skip_before_action(:authenticate)
 
       def index
-        render json: Form.search(search_term: params[:query]),
+        if Flipper.enabled?(:new_va_forms_search)
+          if params[:query].present?
+            if params[:query].match(/^\d{2}(?:[pP])?[- \s]\d+(?:[a-zA-Z])?$/)
+                             .present? || params[:query]
+               .match(/^[sS][fF](?:[- \s])?\d+(?:[a-zA-Z])?$/)
+               .present?
+              search_by_form_number
+            else
+              search_by_text
+            end
+          else
+            return_all
+          end
+        else
+          old_search
+        end
+      end
```

Comment:
this might be a bit nitpicky but this is a bit difficult for me to read personally.
i think switching this to primarily use guard clauses would help immensely, especially with all these nicely broken out methods already:
https://anthonygharvey.com/ruby/guard_clauses_vs_nested_if_statements

Reply:
Not at all nitpicky, I think that is great advice. I'll get it updated